examples/js/loaders/GLTFLoader.js

```diff
@@ -3182,7 +3182,68 @@ THREE.GLTFLoader = ( function () {
 
 					}
 
-					return node;
+					if ( nodeDef.skin === undefined ) return node;
+
+					var skinEntry;
+
+					return parser.getDependency( 'skin', nodeDef.skin ).then( function ( skin ) {
+
+						skinEntry = skin;
+
+						var pendingJoints = [];
+
+						for ( var i = 0, il = skinEntry.joints.length; i < il; i ++ ) {
+
+							pendingJoints.push( parser.getDependency( 'node', skinEntry.joints[ i ] ) );
+
+						}
+
+						return Promise.all( pendingJoints );
+
+					} ).then( function ( jointNodes ) {
+
+						var meshes = node.isGroup === true ? node.children : [ node ];
+
+						for ( var i = 0, il = meshes.length; i < il; i ++ ) {
+
+							var mesh = meshes[ i ];
+
+							var bones = [];
+							var boneInverses = [];
+
+							for ( var j = 0, jl = jointNodes.length; j < jl; j ++ ) {
+
+								var jointNode = jointNodes[ j ];
+
+								if ( jointNode ) {
+
+									bones.push( jointNode );
+
+									var mat = new THREE.Matrix4();
+
+									if ( skinEntry.inverseBindMatrices !== undefined ) {
+
+										mat.fromArray( skinEntry.inverseBindMatrices.array, j * 16 );
+
+									}
+
+									boneInverses.push( mat );
+
+								} else {
+
+									console.warn( 'THREE.GLTFLoader: Joint "%s" could not be found.', skinEntry.joints[ j ] );
+
```

Comment:
Any idea why we have the outer IIFE (starting with the `return ( function() {...` on line 3132? This is nested deeply enough that it's hard to read for me; removing the IIFE might help, or perhaps moving some of the skin-related code into a helper function here.

Reply:
Seems like from #15350